typing/jkind_types.mli

```diff
@@ -106,53 +106,61 @@ module Layout : sig
   end
 end
 
-module Baggage : sig
+module With_bounds : sig
+  module Type_info : sig
+    type +'type_expr t =
+      { type_expr : 'type_expr;
+        modality : Mode.Modality.Value.Const.t;
+        nullability : bool
+      }
+
+    val create :
+      type_expr:'type_expr ->
+      modality:Mode.Modality.Value.Const.t ->
+      deep_only:bool ->
```

Comment:
Unclear what `deep_only` means now -- document?

Reply:
I kinda wanna just rename it to `nullability_matters` or something like that. wdyt?

Reply:
(did that)